vault_service/views/user.py

```diff
@@ -195,4 +197,353 @@ def store_data():
+
+            output = {'id': setup.id,
+                      'name': setup.name,
+                      'qid': qid,
+                      'type': setup.type,
+                      'active': setup.active,
+                      'stateful': setup.stateful,
+                      'frequency': setup.frequency,
+                      'template': setup.template,
+                      'classes': setup.classes,
+                      'data': setup.data,
+                      'created': setup.created.isoformat(),
+                      'updated': setup.updated.isoformat()}
+
+        return json.dumps(output), 200
+
+    elif request.method == 'POST':
+        # check payload
+        try:
+            ntype = payload['type']
+        except KeyError:
+            return json.dumps({'msg': 'No notification type passed'}), 400
+
+        with current_app.session_scope() as session:
+            try:
+                q = session.query(User).filter_by(id=user_id).one()
+            except ormexc.NoResultFound:
+                u = User(id=user_id)
+                try:
+                    session.add(u)
+                    session.commit()
+                except exc.IntegrityError as e:
+                    session.rollback()
+                    return json.dumps({'msg': 'User does not exist, so new myADS setup was not saved, error: {0}'.
+                                      format(e)}), 500
+
+        if ntype == 'query':
+            if not all(k in payload for k in ('qid', 'name', 'stateful', 'frequency')):
+                return json.dumps({'msg': 'Bad data passed'}), 400
+            with current_app.session_scope() as session:
+                qid = payload.get('qid')
+                q = session.query(Query).filter_by(qid=qid).first()
+                if not q:
+                    return json.dumps({'msg': 'Query does not exist'}), 404
+                query_id = q.id
+                setup = MyADS(user_id=user_id,
+                              type='query',
+                              query_id=query_id,
+                              name=payload.get('name'),
+                              active=True,
+                              stateful=payload.get('stateful'),
+                              frequency=payload.get('frequency'))
+
+        elif ntype == 'template':
+            if 'template' not in payload:
+                return json.dumps({'msg': 'Bad data passed'}), 400
+
+            if not isinstance(payload.get('data'), basestring):
+                return json.dumps({'msg': 'Bad data passed'}), 400
+
+            if payload['template'] == 'arxiv':
+                if not isinstance(payload.get('classes'), list):
+                    return json.dumps({'msg': 'Bad data passed'}), 400
+
+            # add metadata
+            if payload['template'] == 'arxiv':
+                template = 'arxiv'
+                classes = payload['classes']
```

Comment:
No, there isn't right now, and I'm not sure how best to handle it. The list of accepted classes is dozens of items long and can change whenever arXiv adds new classes. Do you think it's worth storing this? It can be in the config or something, but it's just a lot to keep track of.

Reply:
Yeah, I'm afraid so, at least I would have a list in the config file. The drawback is that every time we add a new class to BBB, we need to also add it here. But otherwise bad users can store anything in there, which goes against good practices and in particular, we are using a text field which I think it has unlimited size (based on a quick google search).

Reply:
Done. Btw fun fact: there are 166 arxiv classes and subclasses.